templates/vdbench_save.yml

```diff
@@ -0,0 +1,160 @@
+        --resource-group ${AZURE_STORAGE_ACCOUNT_RG_NAME}       \
+        --subscription   $ARM_SUBSCRIPTION_ID --query "[1].value" --output tsv)
+    rc=$?
+    echo "##vso[task.setvariable variable=dest_storacct_key;issecret=true]${dest_storacct_key}"
+    check_rc.sh $rc "Get destination Storage Account key" || exit 0
+
+    dest_container_sas=$(az storage container generate-sas      \
+        --account-key  "${dest_storacct_key}"                   \
+        --account-name ${AZURE_STORAGE_ACCOUNT_NAME}            \
+        --name         ${AZURE_STORAGE_ACCOUNT_CONTAINER_NAME}  \
+        --expiry       "${end_timestamp}"                       \
+        --permissions clrw --https-only --output tsv)
+    rc=$?
+    echo "##vso[task.setvariable variable=dest_container_sas;issecret=true]${dest_container_sas}"
+    check_rc.sh $rc "Generate destination SAS URL" || exit 0
+
+    sas_url="https://${AZURE_STORAGE_ACCOUNT_NAME}.blob.core.windows.net/${AZURE_STORAGE_ACCOUNT_CONTAINER_NAME}?${dest_container_sas}"
+    rc=$?
+    echo "##vso[task.setvariable variable=sas_url;issecret=true]${sas_url}"
+    check_rc.sh $rc "Full SAS URL" || exit 0
+
+    cmdokmsg="CMD_OK_$(date -u '+%m%dx%H%M%S')"
+    cmdlog=/tmp/vm_runcmd_install_azcopy.log
+    az vmss run-command invoke                     \
+      --resource-group ${RESOURCE_GROUP_NAME}-vmss \
+      --name           "vmss"                      \
+      --instance-id    0                           \
+      --command-id     RunShellScript              \
+      --scripts        "runuser -l azureuser -c 'wget https://aka.ms/downloadazcopy-v10-linux && tar -xvf downloadazcopy-v10-linux && sudo cp ./azcopy_linux_amd64_*/azcopy /usr/bin/' && echo ${cmdokmsg}" \
+      --query 'value[0].message' --output tsv |& tee ${cmdlog}
+    grep -q ${cmdokmsg} ${cmdlog} || check_rc.sh 1 "Install azcopy"
+
+    cmdokmsg="CMD_OK_$(date -u '+%m%dx%H%M%S')"
+    cmdlog=/tmp/vm_runcmd_tar.log
+    az vmss run-command invoke                     \
+      --resource-group ${RESOURCE_GROUP_NAME}-vmss \
+      --name           "vmss"                      \
+      --instance-id    0                           \
+      --command-id     RunShellScript              \
+      --scripts        "runuser -l azureuser -c 'tar -c -f ${RESOURCE_GROUP_NAME}.tar /home/azureuser/vdbench/output/' && echo ${cmdokmsg}" \
+      --query 'value[0].message' --output tsv |& tee ${cmdlog}
+    grep -q ${cmdokmsg} ${cmdlog} || check_rc.sh 1 "Tar vdbench output directory"
+
+    cmdokmsg="CMD_OK_$(date -u '+%m%dx%H%M%S')"
+    cmdlog=/tmp/vm_runcmd_azcopy.log
+    az vmss run-command invoke                     \
+      --resource-group ${RESOURCE_GROUP_NAME}-vmss \
+      --name           "vmss"                      \
+      --instance-id    0                           \
+      --command-id     RunShellScript              \
+      --scripts        "azcopy copy '/home/azureuser/${RESOURCE_GROUP_NAME}.tar' '${sas_url}'   && echo ${cmdokmsg}" \
```

Comment:
It'd be helpful to have some sort of traceable link between a pipeline run and the results blob. Maybe that can be accomplished in the blob's name? Or maybe some summary output? Give it some thought on the process of someone evaluating the results and how they'd reconcile that cluster, region, configuration information. Maybe reach out to known potential users internally to understand what would make it efficient for them to ingest this data (and, as a bonus, avoid asking you "where did this data come from?" multiple times :) ).

Reply:
Talked with Bernie, keeping rgname, add vbench config to name.
Changed build number to include these values. 

Testing changes now